### ISSUE TITLE
Rewrites in Middleware do not work on Netlify

### ISSUE THREAD

**Issue by hrishikesh-k** [withastro/astro#11929]
### Astro Info

```block
Astro                    v4.15.3
Node                     v20.17.0
System                   Linux (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/netlify
Integrations             none
```

### Describe the Bug

In the provided reproduction, the following happens:

- using `context.rewrite()` in the middleware returns a 200 status with empty body
- using `next()` in middleware does not rewrite at all
- using `Astro.rewrite()` in the frontmatter of `/page-1` works fine

### What's the expected result?

Rewrite should work fine in all the cases

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-9zs2qt

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Is this issue only happening in production? I notice that the reproduction has `edgeMiddleware: true` defined as well, is that required to see the problem?

**Comment by hrishikesh-k** [comment]
Yeah, it happens only when deployed to production. And it _seems_ to work fine with `edgeMiddleware` disabled. Checking if that's true for all cases...

**Comment by hrishikesh-k** [comment]
Additional data point:

The issue also happens with `edgeMiddleware` disabled if `export const prerender = true`. It works fine if the page is using SSR, but fails with SSG.
